File: arm.64/langs/haskell/src/Boot.hs
module Boot where

import Type
import Find
import Bind
import System.Environment
import Data.Maybe (catMaybes)

boot :: Maybe String -> IO [Cabi]
boot targetDir = do
  envLib <- lookupEnv "PAT_LIB"
  case envLib of
    Just lib | not (null lib) -> do
      m <- bind lib
      case m of
        Just c -> return [c]
        Nothing -> return []
    _ -> do
      files <- Find.find targetDir
      results <- mapM Bind.bind files
      return (catMaybes results)

File: arm.64/langs/python/src/boot.py
import os
from typing import List, Optional
from type import Cabi
from find import find
from bind import bind

def boot(target_dir: Optional[str] = None) -> List[Cabi]:
    engines = []
    env_lib = os.environ.get("PAT_LIB")
    if env_lib:
        c = bind(env_lib)
        if c:
            engines.append(c)
            return engines

    files = find(target_dir)
    for f in files:
        c = bind(f)
        if c:
            engines.append(c)
    return engines
